src/main/java/Parser.java

```diff
@@ -0,0 +1,89 @@
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Scanner;
+
+// deals with making sense of the user command
+public class Parser {
```

Comment:
Your implementation of a Parser seems to not adhere to the Single Responsibility principle. It not only parses commands but also acts on them to modify the taskList and storage. This could be confusing to future developers when first exposed to your codebase